optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/BavetConstraint.java

```diff
@@ -21,42 +21,27 @@
 import java.util.function.Function;
 
 import org.optaplanner.core.api.score.Score;
-import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ScoreImpactType;
 import org.optaplanner.core.impl.score.stream.bavet.common.BavetNodeBuildPolicy;
 import org.optaplanner.core.impl.score.stream.bavet.uni.BavetFromUniConstraintStream;
 import org.optaplanner.core.impl.score.stream.bavet.uni.BavetFromUniNode;
+import org.optaplanner.core.impl.score.stream.common.AbstractConstraint;
 
-public final class BavetConstraint<Solution_> implements Constraint {
+public final class BavetConstraint<Solution_> extends AbstractConstraint<Solution_, BavetConstraintFactory<Solution_>> {
```

Comment:
Both `BavetConstraint` and `DroolsConstraint` get a shared superclass, to get some code reuse in there.